src/Uuid/ModelUuid.php

```diff
@@ -63,21 +63,50 @@ public function id(): string
 		// generate a new ID (to be saved in the content file)
 		$id = static::generate();
 
+		// store the new UUID
+		$this->model = static::storeId($this->model, $id);
+
+		// update the Uri object
+		$this->uri->host($id);
+
+		return $id;
+	}
+
+	/**
+	 * Retrieves the ID string (UUID without scheme) for the model
+	 * from the content file, if it is already stored there
+	 *
+	 * @param \Kirby\Cms\ModelWithContent $model
+	 */
+	public static function retrieveId(Identifiable $model): string|null
+	{
+		return $model->content('default')->get('uuid')->value();
+	}
+
+	/**
+	 * Stores the UUID for the model and makes sure
+	 * to update the content file and content object cache
+	 *
+	 * @param \Kirby\Cms\ModelWithContent $model
+	 * @param string $id
+	 */
+	public static function storeId(Identifiable $model, string $id): Identifiable
+	{
 		// make sure Kirby has the required permissions
 		// for the update action
-		$kirby = $this->model->kirby();
+		$kirby = $model->kirby();
 		$user  = $kirby->auth()->currentUserFromImpersonation();
 		$kirby->impersonate('kirby');
 
 		// get the content array from the page
-		$data = $this->model->content()->toArray();
+		$data = $model->readContent('default');
```

Comment:
Is it necessary to read the content from file here in the "first try"? Wouldn't this work as well:

```suggestion
		$data = $model->content('default')->toArray();
```